### DISCUSSION TITLE
How to emit event on component load

### DISCUSSION THREAD

**Discussion by Padrezz** [livewire/livewire#4082] [1. Help] [answered]
Hey there,

need some help, i'm feeling stuck.

I have distributed code into 3 components. Parent, Child, Modal. 

Parent sends data to children, for example cars, and child split them in his mount() to variables, like brands.

Child{
 public $bmw... etc; 
 public $notsorted; 
}

After split, i need to triger Modal, which recive only "$notsorted" cars and displays message with them. Issue is, that i cannot emit event, from mount. About sorted cars knows only child, which have logic to sort them, so only this component knows to emit data. 

Any idea how to solve this, without users action like click, etc;

**Comment by joshhanley** [comment, marked as answer]
@Padrezz currently you can't emit events from mount.

But you can make use of `wire:init`, see this docs page https://laravel-livewire.com/docs/2.x/defer-loading

Hope this helps!

**Reply by Padrezz** [reply]
Many thanks, looks helpfull :) will try it imidiatly

**Reply by joshhanley** [reply]
No worries 🙂
